node/follower_node.go

```diff
@@ -80,11 +79,15 @@ type AlgorandFollowerNode struct {
 // MakeFollower sets up an Algorand data node
 func MakeFollower(log logging.Logger, rootDir string, cfg config.Local, phonebookAddresses []string, genesis bookkeeping.Genesis) (*AlgorandFollowerNode, error) {
 	node := new(AlgorandFollowerNode)
-	node.rootDir = rootDir
 	node.log = log.With("name", cfg.NetAddress)
 	node.genesisID = genesis.ID()
 	node.genesisHash = genesis.Hash()
 	node.devMode = genesis.DevMode
+	genesisDirs, err := cfg.EnsureAndResolveGenesisDirs(rootDir, genesis.ID())
```

Comment:
not sure I understand why do we have `EnsureAndResolveGenesisDirs` *and* `config.ResolvedGenesisDirs` creation here. Looks like `EnsureAndResolveGenesisDirs` (or a new helper) has all info needed to produce fully initialized `ResolvedGenesisDirs`.
Plus, this would eliminate all code duplication in node.go